contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go

```diff
@@ -165,6 +172,17 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) error
 	}
 	src.Status.MarkDeployed()
 
+	// Only create EventTypes for Broker sinks.
+	if src.Spec.Sink.Kind == "Broker" {
+		err = r.reconcileEventTypes(ctx, src)
+		if err != nil {
+			logger.Error("Unable to reconcile the event types", zap.Error(err))
+			return err
+		}
+	}
+	// We mark EventTypes in order to have the source Ready, even though the Sink might haven't been a Broker.
+	src.Status.MarkEventTypes()
```

Comment:
I recommend removing `EventTypesProvided` from the set of conditions that indicate Ready. It should still be a condition, but it shouldn't be part of source readiness.

The registry is optional. The code here acknowledges this by setting `EventTypesProvided` true when the sink is not a Broker, but Brokers may have an accept everything policy, which should seemingly produce the same behavior.

I think it's clearer for the user to define source readiness as "capable of sending events". If the sink is rejecting events, that's a problem (and possibly an error condition), but the source is still ready.

Reply:
Makes total sense! Thanks for the comment. Will do that from the other controllers I changed as well.